synapse/storage/databases/main/devices.py

```diff
@@ -1747,9 +1747,6 @@ def _update_remote_device_list_cache_entry_txn(
                 table="device_lists_remote_cache",
                 keyvalues={"user_id": user_id, "device_id": device_id},
                 values={"content": json_encoder.encode(content)},
-                # we don't need to lock, because we assume we are the only thread
-                # updating this user's devices.
-                lock=False,
```

Comment:
Handled by unsafe to upsert mechanism: https://github.com/matrix-org/synapse/blob/c3a4780080a5bcb04132283c0f32f7452655792a/synapse/storage/database.py#L93